tests/flag_util_test.py

```diff
@@ -132,5 +133,24 @@ def testReadAndWrite(self):
     self.assertFlagState(flag_values_copy, 2, True)
 
 
+class TestUnitsParser(unittest.TestCase):
+  def setUp(self):
+    self.up = flag_util.UnitsParser()
+
+  def testParser(self):
+    self.assertEqual(self.up.Parse('10KiB'),
+                     10 * 1024 * perfkitbenchmarker.UNIT_REGISTRY.bytes)
+
+  def testQuantity(self):
+    quantity = 1.0 * perfkitbenchmarker.UNIT_REGISTRY.meter
+
+    self.assertEqual(self.up.Parse(quantity),
+                     quantity)
+
+  def testBadExpression(self):
+    with self.assertRaises(ValueError):
+      self.up.Parse('asdf')
```

Comment:
In the old version, nothing, because the UnitParser is supposed to be agnostic about what unit it's parsing. In the code I'm about to upload, I added a feature where the user can specify that the input must be convertible to some specific unit, which you can use to verify that, for instance, your input is convertible to bytes. This lets you accept MB, GB, etc., without accepting meters. (This is a hack around the fact that you can't directly ask pint whether a quantity is in a specified dimension, even though pint's concept of dimensions is exactly what we need here.)